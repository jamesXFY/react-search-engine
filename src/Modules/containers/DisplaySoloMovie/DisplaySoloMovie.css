.soloMovieContainer{
    min-height: 500px;
    align-content: center;
    /* align-items: center; */
    /* text-align: center; */
}
.movie-loading{
    position: absolute;
    top: 50%;
    left:47%;
}