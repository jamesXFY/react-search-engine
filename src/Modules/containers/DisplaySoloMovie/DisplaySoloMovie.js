import React, { Component } from 'react';

import { Container, Row, Col } from 'react-bootstrap';
import SoloMovie from '../../components/SoloMovieComponent/SoloMovie';

import './DisplaySoloMovie.css'


const endPointProps = {
    apiKey: "6ed12e064b90ae1290fa326ce9e790ff",
    apiURL: "https://api.themoviedb.org/3"
}


class DisplaySoloMovie extends Component {
    state = {
        movieId: '',
        title: '',
        movie: {},
        isLoading: true,
    }

    componentDidMount() {
        if(this.props.movieId){
            this.loadMovie();
        }
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevState.movieId !== this.state.movieId) {
            this.loadMovie();
        }
    }

    // we use a timeout to prevent the api request to fire immediately as we type
    timeout = null;

    loadMovie = () => {
        this.setState({ movieId: this.props.movieId })
        clearTimeout(this.timeout);
        this.timeout = setTimeout(() => {
            fetch(endPointProps.apiURL + '/movie/' + this.props.movieId + '?api_key=' + endPointProps.apiKey + '&language=en-US')
                .then(res => res.json())
                .then(
                    (result) => {
                        console.info(result);
                        this.setState({ movie: result, isLoading: false });
                        console.info(this.state);
                    },
                    (error) => {
                        console.info(error);
                    }
                )
        }, 1000)
    }

    backToMain = () => {
        this.props.backToMain();
    }

    render() {
        return (
            <Container>
                <Row>
                    <Col className="soloMovieContainer">
                        {this.state.isLoading && this.props.movieId ?
                            <div class="spinner-border text-primary movie-loading" role="status">
                                <span class="sr-only">Loading...</span>
                            </div>
                            :
                            <SoloMovie movie={this.state.movie} backToMain={this.backToMain}></SoloMovie>
                        }
                    </Col>
                </Row>
            </Container>
        );
    }
}

export default DisplaySoloMovie;