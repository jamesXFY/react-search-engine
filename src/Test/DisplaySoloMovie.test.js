import React from 'react';
import DisplaySoloMovie from '../Modules/containers/DisplaySoloMovie/DisplaySoloMovie';
import { configure, mount, shallow, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });


const movieId = "166428";




describe('DisplaySoloMovie', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<DisplaySoloMovie movieId={movieId} debug/>);
    expect(component).toMatchSnapshot();
  });
});

it('should render correctly with no props', () => {
  const component = shallow(<DisplaySoloMovie/>);
  expect(component).toMatchSnapshot();
});

it('should render correctly with props', () => {
  const component = shallow(<DisplaySoloMovie movieId={movieId}/>);
  expect(component).toMatchSnapshot();
});