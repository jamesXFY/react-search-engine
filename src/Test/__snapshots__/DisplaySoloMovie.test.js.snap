// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplaySoloMovie should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplaySoloMovie
    debug={true}
    movieId="166428"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": <ForwardRef(render)>
        <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
        </ForwardRef>
      </ForwardRef(render)>,
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>,
          "className": "soloMovieContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              class="sr-only"
            >
              Loading...
            </span>,
            "class": "spinner-border text-primary movie-loading",
            "role": "status",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
              "class": "sr-only",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "span",
          },
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": <ForwardRef(render)>
          <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </ForwardRef>
        </ForwardRef(render)>,
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>,
            "className": "soloMovieContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                class="sr-only"
              >
                Loading...
              </span>,
              "class": "spinner-border text-primary movie-loading",
              "role": "status",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
                "class": "sr-only",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "span",
            },
            "type": "div",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplaySoloMovie />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": <ForwardRef(render)>
        <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <SoloMovie
            backToMain={[Function]}
            movie={Object {}}
          />
        </ForwardRef>
      </ForwardRef(render)>,
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <SoloMovie
            backToMain={[Function]}
            movie={Object {}}
          />
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <SoloMovie
            backToMain={[Function]}
            movie={Object {}}
          />,
          "className": "soloMovieContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "backToMain": [Function],
            "movie": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": <ForwardRef(render)>
          <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <SoloMovie
              backToMain={[Function]}
              movie={Object {}}
            />
          </ForwardRef>
        </ForwardRef(render)>,
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <SoloMovie
              backToMain={[Function]}
              movie={Object {}}
            />
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <SoloMovie
              backToMain={[Function]}
              movie={Object {}}
            />,
            "className": "soloMovieContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "backToMain": [Function],
              "movie": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`should render correctly with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DisplaySoloMovie
    movieId="166428"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "as": "div",
      "children": <ForwardRef(render)>
        <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
        </ForwardRef>
      </ForwardRef(render)>,
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          as="div"
          className="soloMovieContainer"
        >
          <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>
        </ForwardRef>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <div
            class="spinner-border text-primary movie-loading"
            role="status"
          >
            <span
              class="sr-only"
            >
              Loading...
            </span>
          </div>,
          "className": "soloMovieContainer",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              class="sr-only"
            >
              Loading...
            </span>,
            "class": "spinner-border text-primary movie-loading",
            "role": "status",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Loading...",
              "class": "sr-only",
            },
            "ref": null,
            "rendered": "Loading...",
            "type": "span",
          },
          "type": "div",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
          },
          "displayName": "Col",
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": undefined,
        "displayName": "Bootstrap(Row)",
        "propTypes": undefined,
        "render": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "as": "div",
        "fluid": false,
      },
      "displayName": "Container",
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "as": "div",
        "children": <ForwardRef(render)>
          <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </ForwardRef>
        </ForwardRef(render)>,
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <ForwardRef
            as="div"
            className="soloMovieContainer"
          >
            <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>
          </ForwardRef>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <div
              class="spinner-border text-primary movie-loading"
              role="status"
            >
              <span
                class="sr-only"
              >
                Loading...
              </span>
            </div>,
            "className": "soloMovieContainer",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span
                class="sr-only"
              >
                Loading...
              </span>,
              "class": "spinner-border text-primary movie-loading",
              "role": "status",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Loading...",
                "class": "sr-only",
              },
              "ref": null,
              "rendered": "Loading...",
              "type": "span",
            },
            "type": "div",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
            },
            "displayName": "Col",
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "displayName": "Bootstrap(Row)",
          "propTypes": undefined,
          "render": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "defaultProps": Object {
          "as": "div",
          "fluid": false,
        },
        "displayName": "Container",
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
